server/__tests__/integration/representatives.test.js

```diff
@@ -15,17 +18,127 @@ afterAll(async () => {
   await new Promise((resolve) => setTimeout(() => resolve(), 500)) // avoid jest open handle error
 })
 
-describe('/api/representatives/:zipcode', () => {
-  test('returns 200 status', async () => {
-    const response = await request(app).get('/api/representatives/92107')
-    expect(response.status).toBe(200)
-  })
-  test('returns 200 status', async () => {
-    const response = await request(app).get('/api/representatives/92107-1234')
+describe('GET  api/representatives/:zipCode', () => {
+  const zipCode = '20510-1234'
+  const route = `/api/representatives/${zipCode}`
+
+  const exampleZipResponse = {
+    errors: [],
+    messages: [],
+    results: {
+      candidates: [
+        {
+          count: {},
+          officials: [],
+          locator: 'World',
+          score: 100,
+          match_addr: ` ${zipCode}, Philadelphia, Pennsylvania`,
+          x: -75.15263499999998,
+          y: 39.93611000000004,
+          wkid: 4326,
+          locator_type: 'Postal',
+          geoservice: 'EsriWGS',
+          match_city: 'Philadelphia',
+          match_subregion: 'Philadelphia County',
+          match_region: 'PA',
+          match_postal: zipCode,
+          match_country: 'US',
+          match_streetaddr: ''
+        }
+      ]
+    }
+  }
+  test('returns 200 status for a a successful mock enpoint', async () => {
+    const spy = jest.spyOn(axios, 'get')
+    spy.mockImplementation((url) => {
+      if (url === `${CICERO_API_HOST}?search_postal=${zipCode.slice(0, 5)}`) {
```

Comment:
@rcmtcristian @thyeggman I think there are supposed to be other query parameters in addition to this one, so the net URL would look a little different than what you are equating it to, here. See [this](https://github.com/ProgramEquity/amplify/blob/main/server/routes/api/representatives.js#L74-L82) for the possible parameters.
```js
`${CICERO_API_HOST}?search_postal=${zipCode.slice(0, 5)}&search_country=US&order=district_type&sort=asc&max=200&format=json&key=${CICERO_API_KEY}`
```

The order might still be different here if you are parsing the actual URL, so I would recommend you use `qs` to manually parse the URL.

Additionally, I don't think the mocked implementation here is correct, because `axios.get` accepts [2 parameters](https://github.com/ProgramEquity/amplify/blob/main/server/routes/api/representatives.js#L88-L99) instead of just a `url`, where the URL is simply `CICERO_API_HOST` and the second parameter contains all of the parameters, and a couple more things. This means you don't actually need to parse the final URL. You already have access to the parameters from the second argument itself.
```diff
-    spy.mockImplementation((url) => {
+    spy.mockImplementation((url, { params }) => {
```